test/ttmlir/Dialect/TTNN/sharding_matmul_0.mlir

```diff
@@ -0,0 +1,14 @@
+// RUN: ttmlir-opt --ttir-to-ttnn-backend-pipeline="enable-optimizer=true sharding-pass-enabled=true max-legal-layouts=0" %s | FileCheck %s
+#any_device_tile = #tt.operand_constraint<dram|l1|tile|any_device_tile>
+module attributes {} {
+  func.func @forward(%arg0: tensor<64x128xbf16>, %arg1: tensor<128x96xbf16>, %arg2: tensor<96x64xbf16>) -> tensor<64x64xbf16> {
+    // CHECK: #[[LAYOUT_7:.*]] = #tt.layout<(d0, d1) -> (d0, d1), undef, <8x8>, memref<1x1x!tt.tile<32x32, bf16>, #dram>, interleaved>
+    %0 = tensor.empty() : tensor<64x96xbf16>
+    // CHECK: %[[C:.*]] = "ttnn.matmul"[[C:.*]] -> tensor<64x96xbf16, #[[LAYOUT_7]]>
```

Comment:
Evo tho we use this `[[C:.*]]` in most tests :), it seems to create the variable `C` which is not needed. Try using `{{.*}}` to match anything instead.